cpp/include/cuvs/neighbors/cagra.h

```diff
@@ -378,6 +379,12 @@ cuvsError_t cuvsCagraBuild(cuvsResources_t res,
                            DLManagedTensor* dataset,
                            cuvsCagraIndex_t index);
 
+cuvsError_t cuvsCagraBuildWithIVFPQ(cuvsResources_t res,
```

Comment:
Rather than a separate function - could we just add the ivf-pq params to the cuvsCagraIndexParams object and then just expose the single cuvsCagraBuild function ?

We already have the `cuvsCagraGraphBuildAlgo` enum to specify building with ivf-pq in `cuvsCagraBuild` - we just need to add the ivf-pq params object so that we don't use the default.

Reply:
How do you want me to express the optional nature here? By defaulting to a NULL pointer?

Reply:
@benfred I reworked the PR